src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package integrationTests.mojo.environmentSetup;
+
+import org.apache.maven.plugin.logging.SystemStreamLog;
+import org.junit.rules.TemporaryFolder;
+import org.mule.tools.client.agent.AgentClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.junit.Assert.fail;
+
+public class AgentSetup {
+
+  private Logger log;
+  private static final long TEN_SECONDS = 10000;
+  private static final String MULE_HOME_FOLDER_PREFIX = "/mule-enterprise-standalone-";
+  private static final String MULE_VERSION = "3.8.1";
+  private static final String AGENT_JKS_RELATIVE_PATH = "/conf/mule-agent.jks";
+  private static final String AGENT_YMS_RELATIVE_PATH = "/conf/mule-agent.yml";
+  private static final String EXECUTABLE_FOLDER_RELATIVE_PATH = "/bin/mule";
+  private static final String AMC_SETUP_RELATIVE_FOLDER = "/bin/amc_setup";
+  private static final String UNENCRYPTED_CONNECTION_OPTION = "-I";
+  private static final int NORMAL_TERMINATION = 0;
+  private static final String START_AGENT_COMMAND = "start";
+  private static final String ANCHOR_FILE_RELATIVE_PATH = "/apps/agent-anchor.txt";
+  private static final String STOP_AGENT_COMMAND = "stop";
+  private static String muleExecutable;
+  private static String[] commands;
+  private static String muleHome;
+  private static Runtime runtime;
+  private static Process applicationProcess;
+
+  public AgentSetup() {
+    log = LoggerFactory.getLogger(this.getClass());
+  }
+
+
+  public void start() throws IOException, InterruptedException {
+    Path currentRelativePath = Paths.get("");
+    String targetFolder = currentRelativePath.toAbsolutePath().toString() + File.separator + "target";
+
+    muleHome = targetFolder + MULE_HOME_FOLDER_PREFIX + MULE_VERSION;
+
+    deleteFile(muleHome + AGENT_JKS_RELATIVE_PATH);
+    deleteFile(muleHome + AGENT_YMS_RELATIVE_PATH);
+
+    muleExecutable = muleHome + EXECUTABLE_FOLDER_RELATIVE_PATH;
+
+    runtime = Runtime.getRuntime();
+    commands = new String[2];
+
+    unpackAgent();
+    startMule();
+    checkAgentIsAcceptingDeployments();
+  }
+
+  public void stop() throws IOException, InterruptedException {
+    stopMule();
+    killMuleProcesses();
+  }
+
+  private void checkAgentIsAcceptingDeployments() throws InterruptedException, IOException {
+    int tries = 0;
+    boolean acceptingDeployments;
+    do {
+      TemporaryFolder folder = new TemporaryFolder();
+      File dummyFile = folder.newFile("dummy.zip");
+      AgentClient agentClient = new AgentClient(new SystemStreamLog(), "http://localhost:9999/");
```

Comment:
Not really, now we hard coded it as 9999 as the default mule-standalone comes set